src/storage/remote/protocol.c

```diff
@@ -92,47 +93,120 @@ storageRemoteFilterGroup(IoFilterGroup *filterGroup, const Variant *filterList)
 /***********************************************************************************************************************************
 Callback to write info list into the protocol
 ***********************************************************************************************************************************/
+typedef struct StorageRemoteProtocolInfoListCallbackData
+{
+    MemContext *memContext;                                         // Mem context used to store values from last call
+    PackWrite *write;                                               // Pack to write into protocol
+    time_t timeModifiedLast;                                        // timeModified from last call
+    mode_t modeLast;                                                // mode from last call
+    uid_t userIdLast;                                               // userId from last call
+    gid_t groupIdLast;                                              // groupId from last call
+    String *user;                                                   // user from last call
+    String *group;                                                  // group from last call
+} StorageRemoteProtocolInfoListCallbackData;
+
 // Helper to write storage info into the protocol. This function is not called unless the info exists so no need to write exists or
 // check for level == storageInfoLevelExists.
+//
+// Fields that do no change from one call to the next are omitted to save bandwidth.
 static void
-storageRemoteInfoWrite(ProtocolServer *server, const StorageInfo *info)
+storageRemoteInfoWrite(StorageRemoteProtocolInfoListCallbackData *data, const StorageInfo *info)
 {
     FUNCTION_TEST_BEGIN();
-        FUNCTION_TEST_PARAM(PROTOCOL_SERVER, server);
+        FUNCTION_TEST_PARAM_P(VOID, data);
         FUNCTION_TEST_PARAM(STORAGE_INFO, info);
     FUNCTION_TEST_END();
```

Comment:
Two things:
1) Can *data or *info be NULL? If not, maybe need ASSERTs?
2) Why is FUNCTION_TEST_PARAM(STORAGE_INFO, info); not with an _P:  FUNCTION_TEST_PARAM_P(STORAGE_INFO, info);

Reply:
1. Asserts added.
2. Because STORAGE_INFO is a defined logging type (see storage/info.h) that happens to be a pointer but we don't need to know that here. void * is not a define logging type so we need to specify _P to let the macro know it is a pointer.